Attempt at addressing some feedback from Aaron (#91)
oauthstuff/draft-oauth-rar: main.md

## Token Introspection {#token_introspection}

> Token introspection [@!RFC7662] provides a means for an RS to query the AS to determine information about an access token. It is RECOMMENDED that the applicable authorization detail information be conveyed with `authorization_details` as a top-level member of the introspection response JSON. The `authorization_details` member MUST contain the same structure defined in (#authz_details), potentially filtered and extended for the RS making the introspection request.

Reviewer: As a side remark, https://datatracker.ietf.org/doc/html/draft-ietf-oauth-jwt-introspection-response-12 comes along with RFC7662, and in its case, `authorization_details` won't be a top-level member of the introspection response.

Reviewer 2: It's a top level member of the token_introspection claim in that case, which is the token introspection response.

> Token introspection [@!RFC7662] provides a means for an RS to query the AS to determine information about an access token. It is RECOMMENDED that the applicable authorization detail information be conveyed with `authorization_details` as a top-level member of the introspection response JSON. The `authorization_details` member MUST contain the same structure defined in (#authz_details), potentially filtered and extended for the RS making the introspection request.

Reviewer 3: I think instead we want to say that if you're going to do it, you have to do it exactly this way:

```suggestion
Token introspection [@!RFC7662] provides a means for an RS to query the AS to determine information about an access token. If the AS includes authorization details for the token in its response, the applicable authorization detail information MUST be conveyed with `authorization_details` as a top-level member of the introspection response JSON object. The `authorization_details` member MUST contain the same structure defined in (#authz_details), potentially filtered and extended for the RS making the introspection request.
```

# Metadata {#metadata}

> Authorization servers SHOULD publish the supported list of authorization details types using the metadata parameter `authorization_details_types_supported`, which is a JSON array.

Reviewer 3: Similar to above, if it's included it MUST be in this field.

```suggestion
If the AS wants to advertise its support for this feature, the supported list of authorization details types MUST be included in the AS metadata response [@!RFC8414] using the metadata parameter `authorization_details_types_supported`, which is a JSON array.
```

# Privacy Considerations {#privacy_considerations}

> Implementers MUST design and use authorization details in a privacy-preserving manner.

Reviewer 3: We're not supposed to have normative requirements in the privacy section.

```suggestion
It is especially important for implementers to design and use authorization details in a privacy-preserving manner.
```

Author: There are three other MUST occurrences in Privacy Considerations that should probably be changed too. 

I've never been sure if not supposed to have normative requirements in the privacy section a hard and fast rule. Or recommendation. Or rumor.

Author: There's a SHOULD in there now too...

Author: I just went ahead and replaced normative keywords with regular language in the privacy considerations 9d57d65
